pyvrp/Population.py

```diff
@@ -156,10 +160,9 @@ def select():
 
             return self._infeas[idx - num_feas]
 
-        item1 = select()
-        item2 = select()
+        if k <= 0:
+            raise ValueError(f"Expected k > 0; got k = {k}.")
 
-        if item1.fitness < item2.fitness:
-            return item1.individual
-
-        return item2.individual
+        items = [select() for _ in range(k)]
```

Comment:
We could pass in a generator to min directly?

Reply:
Yes, but I'm reasonably sure that is slower. There also isn't a real memory concern here (because a reasonable value for $k$ is fairly small, say, <10).